src/com/amazon/corretto/crypto/provider/ConstantTime.java

```diff
@@ -51,4 +53,44 @@ static final int select(int selector, int a, int b) {
 
     return b ^ (combined & mask);
   }
+
+  /**
+   * @return true iff all the bytes in the specified ranges are equal
+   */
+  private static boolean equals(
+      final byte[] a,
+      final int aStart,
+      final int aLen,
+      final byte[] b,
+      final int bStart,
+      final int bLen) {
+
+    Utils.checkArrayLimits(a, aStart, aLen);
+
+    if (aLen != bLen) {
+      return false;
+    }
+
+    int result = 0;
+
+    for (int i = 0; i < aLen; i++) {
+      result |= a[aStart + i] ^ b[bStart + i];
+    }
+
+    return result == 0;
+  }
+
+  static boolean equals(final byte[] a, final int aStart, final int aLen, final byte[] b) {
```

Comment:
nit: do we want to do null checks here too? right now we'd NPE on L84 if `b` is null

Reply:
It's a method used internally, but I'll add the check. We should invest in CheckerFramework to avoid these runtime checks for internal methods and statically ensure that all use cases pass none null values.